pkg/discovery/network/pods.go

```diff
@@ -55,8 +55,13 @@ func findPod(clientSet kubernetes.Interface, labelSelector string) (*v1.Pod, err
 		Limit:         1,
 	})
 
-	if err != nil || len(pods.Items) == 0 {
+	if err != nil {
 		return nil, errors.WithMessagef(err, "error listing Pods by label selector %q", labelSelector)
```

Comment:
`WithMessagef` actually returns `nil` if `err = nil`, but it is confusing. So, let's separate `err != nil` case and the other case.